Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -870,12 +870,22 @@ lemma Nil.append {p : G.Walk u v} {q : G.Walk v w} (hp : p.Nil) (hq : q.Nil) : (
 lemma nil_reverse {p : G.Walk v w} : p.reverse.Nil ↔ p.Nil := by
   cases p <;> simp
 
+@[simp] lemma nil_copy {x y x' y' : V} {p : G.Walk x y} (hx : x = x') (hy : y = y') :
+    (p.copy hx hy).Nil = p.Nil := by
+  subst_vars; rfl
+
 /-- A walk with its endpoints defeq is `Nil` if and only if it is equal to `nil`. -/
 lemma nil_iff_eq_nil : ∀ {p : G.Walk v v}, p.Nil ↔ p = nil
   | .nil | .cons _ _ => by simp
 
 alias ⟨Nil.eq_nil, _⟩ := nil_iff_eq_nil
 
+lemma eq_nil_copy_of_nil {p : G.Walk u v} (h : p.Nil) :
+    p = nil.copy h.eq.symm rfl := by
```

Comment:
I think the rhs of this is making the nil walk from v to v, and copying it to a walk from u to v, right? Is it any nicer to make it as the walk from u to u, then changing the rhs. ie `nil.copy rfl h.eq`? That would propagate to `drop_length_of_le` etc